### ISSUE TITLE
Bug with hybrid pair styles when atoms are added to the simulation

### ISSUE THREAD

**Issue by akohlmey** [lammps/lammps#465]
From lammps-users,

this may or may not be related to issue #462 
full e-mail thread on lammps-users is at: https://sourceforge.net/p/lammps/mailman/message/35769753/
example inputs for testing are linked here:
[neighborbug.tar.gz](https://github.com/lammps/lammps/files/964677/neighborbug.tar.gz)
[testlist.tar.gz](https://github.com/lammps/lammps/files/964678/testlist.tar.gz)

```
Dear List,

I'm working on generalizing the fix_bond_swap to work for the mSRP
polymer model. This is horribly tricky, and my code keeps crashing.
Hence I started to valgrind to check for memory problems. I get some
errors that persist even if I remove most of my code, keeping only the
skeleton code to request and build a full neighbor list.

Using the in.srp example and LAMMPS dated 31/3 running on a single core
compiled with gcc -O0 -ggdb, valgrind does not report any memory issues
except a few leaks. To the source code, I have added the dummy testtest
fix (source below). The dummy fix requests a full pair list, and and
does nothing except call neighbor->build_one(fulllist,1) in postintegrate.

When I add "fix test all testtest 50" to the in.srp script instantiate
the dummy fix. I get the valgrind warnings below, ending with a
segmentation fault.

Now if I instead do a "run 0" before instantiating the dummy fix, then
it runs without any errors.

This suggests to me that there is some subtle bug at play somewhere in
the neighbor code that is triggered by this combination of force field
and neighbor list requests.
```
and 
```
Addendum.

The neighbor build error seems specific to simulations that adds
particles during a simulation.

To check if this was DPD/SRP specific, I used in.srp as a model and
added 1000 randomly distributed type 2 particles to approximate the
DPD/SRP systems, but replacing pair interactions with

pair_style      hybrid dpd 1.0 1.0 373692 zero 0.8
pair_coeff      1 1 dpd 60.0 4.5 1.0
pair_coeff      1 2 none
pair_coeff      2 2 zero
neigh_modify exclude type 1 2

and added fixes and computes to check all neighbor lists combinations.
This runs fine without valgrind reporting any memory issues.

Now using the same pair style as above, but inserting the particles
during the simulation with

fix dep all deposit 10 2 10 235245 region box

Again all computes and fixes did invalid writes. 
```

**Comment by akohlmey** [comment]
@sjplimp just checked this input deck again and it is still failing, i.e. the existing bugfixes to neighbor lists don't address this issue.

**Comment by stanmoore1** [comment]
Right now occasional lists are never grown except once at the beginning, so if atoms are added then there are invalid reads/writes. Adding this change makes it Valgrind clean for the test I ran:

<pre>
$ svn diff
Index: neighbor.cpp
===================================================================
--- neighbor.cpp        (revision 16493)
+++ neighbor.cpp        (working copy)
@@ -2193,6 +2193,11 @@

   // build the list

+
+  int nlocal = atom->nlocal;
+  int nall = nlocal + atom->nghost;
+
+  if (!mylist->copy) mylist->grow(nlocal,nall);
   np->build_setup();
   np->build(mylist);
 }
</pre>

**Comment by stanmoore1** [comment]
@sjplimp may have a different fix though, so I'll wait to commit until I talk with him

**Comment by stanmoore1** [comment]
I talked to @sjplimp and then I committed this fix to the SVN branch.

**Comment by stanmoore1** [comment]
Fixed by #488. Closed #465.
